sys-block/gpart/patches/gpart-0.3.patchset.2

```diff
@@ -105,51 +120,24 @@ index 3b44a75..71aef4a 100644
  	start = skipsec ? skipsec : d->d_dg.d_s;
 +#endif
  	d->d_nsb = start - incr;
-+
  	start *= d->d_ssize;
  	if (l64seek(d->d_fd, start, SEEK_SET) == -1)
- 		pr(FATAL, EM_SEEKFAILURE, d->d_dev);
 diff --git a/src/gpart.h b/src/gpart.h
-index d973d65..3b59e35 100644
+index d973d65..e9a6e67 100644
 --- a/src/gpart.h
 +++ b/src/gpart.h
-@@ -26,7 +26,35 @@
- 
- typedef uint8_t byte_t;
+@@ -22,6 +22,11 @@
+ #include "errmsgs.h"
+ #include "l64seek.h"
  
--
 +#ifdef __HAIKU__
-+#	if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
-+#		define htole16(x) (x)
-+#		define htobe16(x) __builtin_bswap16(x)
-+#		define htole32(x) (x)
-+#		define htobe32(x) __builtin_bswap32(x)
-+#		define htole64(x) (x)
-+#		define htobe64(x) __builtin_bswap64(x)
-+#		define le16toh(x) (x)
-+#		define be16toh(x) __builtin_bswap16(x)
-+#		define le32toh(x) (x)
-+#		define be32toh(x) __builtin_bswap32(x)
-+#		define le64toh(x) (x)
-+#		define be64toh(x) __builtin_bswap64(x)
-+#	else
-+#		define htole16(x) __builtin_bswap16(x)
-+#		define htobe16(x) (x)
-+#		define htole32(x) __builtin_bswap32(x)
-+#		define htobe32(x) (x)
-+#		define htole64(x) __builtin_bswap64(x)
-+#		define htobe64(x) (x)
-+#		define le16toh(x) __builtin_bswap16(x)
-+#		define be16toh(x) (x)
-+#		define le32toh(x) __builtin_bswap32(x)
-+#		define be32toh(x) (x)
-+#		define le64toh(x) __builtin_bswap64(x)
-+#		define be64toh(x) (x)
-+#	endif
++#define _BSD_SOURCE
++#include <endian.h>
```

Comment:
Please don't do it like this. Instead add "-D_BSD_SOURCE" to your CFLAGS, and only include `<endian.h>` here. `<endian.h>` is present in both BSDs and Linux, so you may not need to put it in `#ifdef __HAIKU__`

Reply:
Actually, looking at btrfs code above, you may not even need to add `<endian.h>`, just add `-D_BSD_SOURCE` to CFLAGS and you're done

Reply:
`<endian.h>` is used elsewhere without being included, still leading to undefined references.
On linux, this is not an error probably because it's included by another header.